extra/innochecksum.cc

```diff
@@ -786,6 +788,16 @@ parse_page(
 
 	/* Check whether page is doublewrite buffer. */
 	str = skip_page ? "Double_write_buffer" : "-";
+	page_no = mach_read_from_4(page + FIL_PAGE_OFFSET);
+	if (skip_freed_pages) {
+		const byte *des= xdes + XDES_ARR_OFFSET +
+			XDES_SIZE * ((page_no & (physical_page_size - 1))
+				     / FSP_EXTENT_SIZE);
+		if (mach_read_from_4(des) != XDES_FSEG &&
+		    xdes_is_free(des, page_no % FSP_EXTENT_SIZE)) {
+			return;
+		}
```

Comment:
`FSP_EXTENT_SIZE` is a rather complex macro that depends on `srv_page_size_shift`. Can we introduce `static size_t extent_size` and assign `extent_size = FSP_EXTENT_SIZE` right after we have determined the page size?